onnxruntime/core/providers/cpu/ml/svmclassifier.cc

```diff
@@ -125,120 +153,71 @@ Status SVMClassifier<T>::Compute(OpKernelContext* ctx) const {
+          for (int64_t m = 0; m < class_i_support_count; ++m, ++val1, ++val2)
+            sum += *val1 * *val2;
+
+          val1 = (float*)&(coefficients_[pos2 + start_index_j]);
+          val2 = (float*)&(kernels[start_index_j]);
+          for (int64_t m = 0; m < class_j_support_count; ++m, ++val1, ++val2)
+            sum += *val1 * *val2;
 
           sum += rho_[evals];
           scores.push_back(sum);
-          if (sum > 0) {
-            votes[i]++;
-          } else {
-            votes[j]++;
-          }
-          evals++;  //index into rho
+          ++(votes[sum > 0 ? i : j]);
+          ++evals;  //index into rho
         }
       }
-    } else if (mode_ == SVM_TYPE::SVM_LINEAR) {     //liblinear
-      for (int64_t j = 0; j < class_count_; j++) {  //for each class
-        float val = kernel_dot(x_data, current_weight_0, coefficients_, feature_count_ * j, feature_count_, get_kernel_type());
-        val += rho_[0];
-        scores.push_back(val);
-      }
     }
+
     if (proba_.size() > 0 && mode_ == SVM_TYPE::SVM_SVC) {
       //compute probabilities from the scores
-      std::vector<float> estimates;
-      std::vector<float> probsp2;
       int64_t num = class_count_ * class_count_;
-      for (int64_t m = 0; m < num; m++) {
-        probsp2.push_back(0.f);  //min prob
-      }
-      for (int64_t m = 0; m < class_count_; m++) {
-        estimates.push_back(0.f);  //min prob
-      }
+      std::vector<float> probsp2(num, 0.f);
+      std::vector<float> estimates(class_count_, 0.f);
       int64_t index = 0;
-      for (int64_t i = 0; i < class_count_; i++) {
-        for (int64_t j = i + 1; j < class_count_; j++) {
+      int64_t p1, p2;
+      for (int64_t i = 0; i < class_count_; ++i) {
+        p1 = i * class_count_ + i + 1;
+        p2 = (i + 1) * class_count_ + i;
+        for (int64_t j = i + 1; j < class_count_; ++j, ++index) {
           float val1 = sigmoid_probability(scores[index], proba_[index], probb_[index]);
           float val2 = std::max(val1, 1.0e-7f);
-          probsp2[i * class_count_ + j] = std::min(val2, 1 - 1.0e-7f);
-          probsp2[j * class_count_ + i] = 1 - probsp2[i * class_count_ + j];
-          index++;
+          val2 = std::min(val2, 1 - 1.0e-7f);
+          probsp2[p1] = val2;
+          probsp2[p2] = 1 - val2;
+          ++p1;
+          p2 += class_count_;
         }
       }
       multiclass_probability(class_count_, probsp2, estimates);
       //copy probabilities back into scores
       scores.resize(estimates.size());
-      for (int64_t k = 0; k < static_cast<int64_t>(estimates.size()); k++) {
-        scores[k] = estimates[k];
-      }
+      std::copy(estimates.begin(), estimates.end(), scores.begin());
+    // } else if (proba_.size() == 0) {
+    // Normalization will be changed in 0.21.
+    // See https://github.com/scikit-learn/scikit-learn/pull/10440
     }
-    int64_t maxvotes = 0;
+
+    double maxweight = 0;
```

Comment:
This name doesn't look great. Could we do `bestClassScore`? Similarly, `bestClassIndex` looks better than `maxClass`.

Reply:
I think we require underscore separation like max_weight at the very least.

---
In reply to: [263863973](https://github.com/Microsoft/onnxruntime/pull/101#discussion_r263863973) [](ancestors = 263863973)

Reply:
Renamed into max_weight

Reply:
Renamed into max_weight